cli/src/pcluster/api/controllers/cluster_logs_controller.py

```diff
@@ -0,0 +1,194 @@
+        return LogEvent.from_dict(event)
+
+    events = [convert_log_event(e) for e in log_events.events]
+    return GetClusterLogEventsResponseContent(
+        events=events, next_token=log_events.next_ftoken, prev_token=log_events.next_btoken
+    )
+
+
+@configure_aws_region()
+@convert_errors()
+def get_cluster_stack_events(cluster_name, region=None, next_token=None):
+    """
+    Retrieve the events associated with the stack for a given cluster.
+
+    :param cluster_name: Name of the cluster
+    :type cluster_name: str
+    :param region: AWS Region that the operation corresponds to.
+    :type region: str
+    :param next_token: Token to use for paginated requests.
+    :type next_token: str
+
+    :rtype: GetClusterStackEventsResponseContent
+    """
+    cluster = Cluster(cluster_name)
+    stack_events = cluster.get_stack_events(next_token=next_token)
+
+    def convert_event(event):
+        event = {k[0].lower() + k[1:]: v for k, v in event.items()}
+        event["timestamp"] = to_iso_time(event["timestamp"])
+        return StackEvent.from_dict(event)
+
+    events = [convert_event(event) for event in stack_events["StackEvents"]]
+    return GetClusterStackEventsResponseContent(next_token=stack_events.get("NextToken"), events=events)
+
+
+@configure_aws_region()
+@convert_errors()
+def list_cluster_log_streams(cluster_name, region=None, filters=None, next_token=None):
+    """
+    Retrieve the list of log streams associated with a cluster.
+
+    :param cluster_name: Name of the cluster
+    :type cluster_name: str
+    :param region: Region that the given cluster belongs to.
+    :type region: str
+    :param filters: Filter the log streams. Format: (Name&#x3D;a,Values&#x3D;1 Name&#x3D;b,Values&#x3D;2,3).
+    :type filters: List[str]
+    :param next_token: Token to use for paginated requests.
+    :type next_token: str
+
+    :rtype: ListClusterLogStreamsResponseContent
+    """
+    filter_parser = _Filter(accepted_filters=["private-dns-name", "node-type"])
+    filters = [filter_parser(f) for f in filters] if filters else None
+    cluster = Cluster(cluster_name)
+    try:
+        if not check_cluster_version(cluster):
+            raise BadRequestException(
+                f"cluster '{cluster_name}' belongs to an incompatible ParallelCluster major version."
+            )
+    except StackNotFoundError:
+        raise NotFoundException(
+            f"cluster '{cluster_name}' does not exist or belongs to an incompatible ParallelCluster major version. "
+        )
+
+    def convert_log(log):
+        log["logStreamArn"] = log.pop("arn")
+        if "storedBytes" in log:
+            del log["storedBytes"]
+        for ts_name in ["creationTime", "firstEventTimestamp", "lastEventTimestamp", "lastIngestionTime"]:
+            log[ts_name] = to_iso_time(log[ts_name])
+        return LogStream.from_dict(log)
+
+    cluster_logs = cluster.list_logs(filters=filters, next_token=next_token)
```

Comment:
are we handling the case when the CW log groups do not exist by returning an empty array?